src/domain/accounts/domain/src/repos/account_repository.rs

```diff
@@ -19,8 +20,14 @@ use crate::Account;
 
 #[async_trait::async_trait]
 pub trait AccountRepository: Send + Sync {
+    // TODO: Private Datasets: tests
+    async fn accounts_count(&self) -> Result<usize, AccountsCountError>;
+
     async fn create_account(&self, account: &Account) -> Result<(), CreateAccountError>;
 
+    // TODO: Private Datasets: tests
+    async fn get_accounts(&self, pagination: PaginationOpts) -> AccountPageStream;
+
```

Comment:
Dangerous. We only need these 2 methods for indexing, but they join the main interface.
Let's prevent the misuse:
 - minimum, by comment - these things should never be called in any public APIs, ...
 - better, by having a separate trait, even if it's implemented in the same struct

Reply:
A separate trait has been added (also an extension for it).

Done